backend/sites/src/app/services/disclosure/disclosure.service.ts

```diff
@@ -7,17 +7,28 @@ import { SiteProfiles } from '../../entities/siteProfiles.entity';
 export class DisclosureService {
   constructor(
     @InjectRepository(SiteProfiles)
-    private disclosureRepository: Repository<SiteProfiles>,
+    private readonly disclosureRepository: Repository<SiteProfiles>,
   ) {}
 
+  /**
+   * Retrieves site profiles for a given site ID.
+   *
+   * @param siteId - The ID of the site whose profiles are to be fetched.
+   * @returns A promise that resolves to an array of SiteProfiles entities.
+   * @throws Error if there is an issue retrieving the data.
+   */
   async getSiteDisclosureBySiteId(siteId: string): Promise<SiteProfiles[]> {
     try {
+      // Fetch site profiles based on the provided siteId
       const result = await this.disclosureRepository.find({
```

Comment:
can result be  null? if yes, please handle

Reply:
Result wont be null, if there will be no data in table the it will return empty array, that is handled at resolver level using this if (result && result.length > 0)